### ISSUE TITLE
Some word documents ingest to elk will cause an WARN message

### ISSUE THREAD

**Issue by xiaoxin01** [elastic/elasticsearch#31572]
**Elasticsearch version** Version: 6.2.2, Build: 10b1edd/2018-02-16T19:01:30.685723Z, JVM: 1.8.0_161

**Plugins installed**: [ik, ingest-attachment]

**OS version** Linux f1154194fb2f 3.19.0-25-generic #26~14.04.1-Ubuntu SMP Fri Jul 24 21:16:20 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

I encoded some word files with base64 encoding and send to elk, some of them may got an WARN message:

```
[2018-06-26T06:49:56,236][WARN ][o.e.m.j.JvmGcMonitorService] [6sJx6Hh] [gc][563] overhead, spent [677ms] collecting in the last [1s]
java.lang.NoSuchMethodException: org.openxmlformats.schemas.wordprocessingml.x2006.main.impl.CTPictureBaseImpl.<init>(org.apache.xmlbeans.SchemaType, boolean)
        at java.lang.Class.getConstructor0(Class.java:3082)
        at java.lang.Class.getDeclaredConstructor(Class.java:2178)
        at org.apache.xmlbeans.impl.schema.SchemaTypeImpl.getJavaImplConstructor2(SchemaTypeImpl.java:1817)
        at org.apache.xmlbeans.impl.schema.SchemaTypeImpl.createUnattachedSubclass(SchemaTypeImpl.java:1961)
        at org.apache.xmlbeans.impl.schema.SchemaTypeImpl.createUnattachedNode(SchemaTypeImpl.java:1950)
        at org.apache.xmlbeans.impl.schema.SchemaTypeImpl.createElementType(SchemaTypeImpl.java:1051)
        at org.apache.xmlbeans.impl.values.XmlObjectBase.create_element_user(XmlObjectBase.java:938)
        at org.apache.xmlbeans.impl.store.Xobj.getUser(Xobj.java:1675)
        at org.apache.xmlbeans.impl.store.Cur.getUser(Cur.java:2659)
        at org.apache.xmlbeans.impl.store.Cur.getObject(Cur.java:2652)
        at org.apache.xmlbeans.impl.store.Cursor._getObject(Cursor.java:995)
        at org.apache.xmlbeans.impl.store.Cursor.getObject(Cursor.java:2904)
        at org.apache.poi.xwpf.usermodel.XWPFDocument.onDocumentRead(XWPFDocument.java:162)
        at org.apache.poi.POIXMLDocument.load(POIXMLDocument.java:169)
        at org.apache.poi.xwpf.usermodel.XWPFDocument.<init>(XWPFDocument.java:112)
        at org.apache.poi.xwpf.extractor.XWPFWordExtractor.<init>(XWPFWordExtractor.java:60)
        at org.apache.poi.extractor.ExtractorFactory.createExtractor(ExtractorFactory.java:243)
        at org.apache.tika.parser.microsoft.ooxml.OOXMLExtractorFactory.parse(OOXMLExtractorFactory.java:105)
        at org.apache.tika.parser.microsoft.ooxml.OOXMLParser.parse(OOXMLParser.java:106)
        at org.apache.tika.parser.ParserDecorator.parse(ParserDecorator.java:188)
        at org.apache.tika.parser.CompositeParser.parse(CompositeParser.java:280)
        at org.apache.tika.parser.AutoDetectParser.parse(AutoDetectParser.java:143)
        at org.apache.tika.Tika.parseToString(Tika.java:568)
        at org.elasticsearch.ingest.attachment.TikaImpl.lambda$parse$0(TikaImpl.java:108)
        at java.security.AccessController.doPrivileged(Native Method)
        at org.elasticsearch.ingest.attachment.TikaImpl.parse(TikaImpl.java:107)
        at org.elasticsearch.ingest.attachment.AttachmentProcessor.execute(AttachmentProcessor.java:88)
        at org.elasticsearch.ingest.common.ForEachProcessor.execute(ForEachProcessor.java:64)
        at org.elasticsearch.ingest.CompoundProcessor.execute(CompoundProcessor.java:100)
        at org.elasticsearch.ingest.Pipeline.execute(Pipeline.java:58)
        at org.elasticsearch.ingest.PipelineExecutionService.innerExecute(PipelineExecutionService.java:169)
        at org.elasticsearch.ingest.PipelineExecutionService.access$000(PipelineExecutionService.java:42)
        at org.elasticsearch.ingest.PipelineExecutionService$2.doRun(PipelineExecutionService.java:94)
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:672)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by markharwood** [comment]
This is an issue related to the Tika dependency which is used to parse binary documents. I suggest you raise an issue on that forum - a quick search has revealed a [similar looking stacktrace](https://issues.apache.org/jira/browse/TIKA-792?jql=project%20%3D%20TIKA%20AND%20text%20~%20nosuchmethodexception)

Generally speaking, when loading old binary documents I wouldn't expect a 100% clean parsing operation due to the many and varied formats out there.

Closing as a non-elasticsearch issue.

**Comment by markharwood** [comment]
Digging deeper - it looks like this is [fixed in POI 4.x ](https://issues.apache.org/jira/browse/TIKA-2549?focusedCommentId=16336164&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16336164) but the ingest-attachment plugin bundles POI 3.17

**Comment by markharwood** [comment]
It looks like the fix to this problem is in POI 4.x but the latest Tika (1.18) only bundles POI 3.17 so we'll have to wait for that fix to work it's way through the TIka release process.

**Comment by tballison** [comment]
Y, see TIKA-2549.  We just released Tika 1.19, which ships with POI 4.0.0, which fixes this.

**Comment by martijnvg** [comment]
This is fixed now, since the attachment plugin uses Tika 1.22 and POI 4.0.1 now.
